proto/wippersnapper/i2c/v1/i2c.proto

```diff
@@ -258,45 +258,29 @@ enum SensorType {
 }
 
 /**
-* SensorDetails is used to describe basic information about a specific sensor.
-*/
-message SensorDetails {
-  string name      = 1; /** The sensor name or ID, up to a maximum of twelve characters (ex. "MPL115A2"). */
-  int32 version    = 2; /** The version of the sensor HW and the driver to allow us to differentiate versions of the board or driver. */
-  int32 id         = 3; /** A unique sensor identifier that is used to differentiate this specific sensor instance from any others that are present on the system or in the sensor network */
-  SensorType type  = 4; /** The sensor type, based on SensorType */
-  float max_value  = 5; /** The maximum value that this sensor can return (in the appropriate SI unit) */
-  float min_value  = 6; /** The minimum value that this sensor can return (in the appropriate SI unit) */
-  float resolution = 7; /** The smallest difference between two values that this sensor can report (in the appropriate SI unit) */
-  int32 min_delay  = 8; /** The minimum delay in microseconds between two sensor events, or '0' if there is no constant sensor rate */
-}
-
-/**
-* SensorEvent  is used to return sensor data from any sensor supported by the abstraction layer,
-* using standard SI units and scales.
+* SensorEvent  is used to return sensor event data from any sensor and includes
+* the SI units of the event data.
 */
 message SensorEvent {
-  int32 version   = 1; /** Contain 'sizeof(sensors_event_t)' to identify which version of the API we're using in case this changes in the future */
-  int32 sensor_id = 2; /** A unique sensor identifier that is used to differentiate this specific sensor instance from any others that are present on the system or in the sensor network */
-  SensorType type = 3; /** The sensor type, based on SensorType */
-  int32 timestamp = 4; /** Time in milliseconds when the sensor value was read */
+  SensorType event_data_unit = 1; /** The event_data's SI unit */
   oneof event_data {
-    float temperature       = 5;
-    float distance          = 6;
-    float light             = 7;
-    float pressure          = 8;
-    float relative_humidity = 9;
-    float current           = 10;
-    float voltage           = 11;
-    uint32 raw_value        = 12;
+    float temperature                 = 2;
+    float distance                    = 3;
+    float light                       = 4;
+    float pressure                    = 5;
+    float relative_humidity           = 6;
+    float current                     = 7;
+    float voltage                     = 8;
+    float raw_value                   = 9;
```

Comment:
Interesting way to do this, I think I would've gone with a pair of fields:
- type int/enum
- raw value float
...but I'm not sure which is better, still getting my head around how I2C components will work end to end.

Also, how/where is it determined which units are being sent?

Reply:
We're doing floats as they can scale w/no precision loss

> Also, how/where is it determined which units are being sent?
The `SensorType` will be set by the device in the I2CDevice driver for the specific I2C sensor.

Reply:
(I missed this response until now, apologies.)

You don't seem to have understood my comment based on your response, I'll restate it explicitly.

Instead of doing one of:
- float a
- float b
- float c

I am suggesting to do:
- int type (tells you if its a, b, or c)
- float value (is the value)

2 fields that are always present instead of a long list of fields that only one is present. Does that make sense? Curious if this solves the problem more elegantly or maybe I'm missing something!